### PR TITLE
CANDLEPIN-820: Removed hibernate criteria from select curator classes

### PR DESCRIPTION
**Pull request by joshmalbrecht**
- Removed hibernate criteria from DistributorVersionCurator, ProductCertificateCurator, and ContentAccessCertificateCurator.
- Added missing java docs to curator methods.
- Added mising unit tests for ContentAccessCertificateCurator and DistributorVersionCurator.

### CODE REVIEW

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java [hunk 1]
```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
...
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves a {@link DistributorVersion} that has an ID equal to the provided ID.
+     *
+     * @param id
+     *  the ID to retrieve a distributor version for
+     *
+     * @return a distributor version that has an ID equal to the provided ID, or null if one does not exist
+     */
     public DistributorVersion findById(String id) {
```
Comment (Ceiu) [inline src/main/java/org/candlepin/model/DistributorVersionCurator.java:119]: Same deal with this method: the universal `get(id)` from AHC can replace this directly.

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java [hunk 2]
```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
...
+     * Retrieves all the {@link DistributorVersionCapability} for a {@link DistributorVersion} that has a
+     * name equal to the provided name.
+     *
+     * @param distVersionName
+     *  the name of the distributor version to retrieve capabilities for
+     *
+     * @return all the capabilities for a distributor version that has a name equal to the provided name, or
+     *  null if there is no distributor version that has the provided name
+     */
+    public Set<DistributorVersionCapability> findCapabilitiesByDistVersion(String distVersionName) {
```
Comment (Ceiu) [inline src/main/java/org/candlepin/model/DistributorVersionCurator.java:160]: This method is only used in one place and could be replaced by fetching the DV straight up and then doing the null check there. Would save maintenance and testing pain here.

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java [hunk 3]
```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
...
-            return dvList.get(0);
+        if (id == null || id.isBlank()) {
+            return null;
+        }
+
+        String query = "SELECT d FROM DistributorVersion d WHERE d.id = :id";
+
+        try {
+            return this.currentSession().createQuery(query, DistributorVersion.class)
+            .setParameter("id", id)
```
Comment (Ceiu) [inline src/main/java/org/candlepin/model/DistributorVersionCurator.java:128]: Indent these chained methods

**Review by Ceiu — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/model/ContentAccessCertificateCurator.java [hunk 4]
```diff
@@ -43,12 +43,32 @@ public ContentAccessCertificateCurator() {
...
+        log.debug("Retrieving content access certificate for consumer: {}",
+            consumer == null ? "null" : consumer.getId());
+        if (consumer == null || consumer.getId() == null || consumer.getId().isBlank()) {
+            return null;
+        }
+
+        String query = "SELECT c FROM SCACertificate c WHERE c.consumer.id = :id";
+
+        try {
+            return this.currentSession().createQuery(query, SCACertificate.class)
```
[the remark this answers is not in the document]
Comment (joshmalbrecht) [reply src/main/java/org/candlepin/model/ContentAccessCertificateCurator.java]: Updated throughout

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java [hunk 5]
```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
...
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves all of the {@link DistributorVersion}s.
+     *
+     * @return
+     *  all distributor version
+     */
     public List<DistributorVersion> findAll() {
```
[the remark this answers is not in the document]
Comment (joshmalbrecht) [reply src/main/java/org/candlepin/model/DistributorVersionCurator.java]: Good catch! Removed

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java (hunk 1 above)
Comment (joshmalbrecht) [reply src/main/java/org/candlepin/model/DistributorVersionCurator.java]: Removed

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java (hunk 2 above)
Comment (joshmalbrecht) [reply src/main/java/org/candlepin/model/DistributorVersionCurator.java]: Good catch! Updated, and then updated the method that called `findCapabilitiesByDistVersion`

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java (hunk 3 above)
Comment (joshmalbrecht) [reply src/main/java/org/candlepin/model/DistributorVersionCurator.java]: Updated

File: src/main/java/org/candlepin/model/DistributorVersionCurator.java [hunk 6]
```diff
@@ -33,66 +31,81 @@ public DistributorVersionCurator() {
...
-        List<DistributorVersionCapability> caps = currentSession()
-            .createCriteria(DistributorVersionCapability.class)
-            .add(Restrictions.eq("name", capability)).list();
-
-        List<DistributorVersion> distVers = new ArrayList<>();
-
-        for (DistributorVersionCapability dvc : caps) {
-            distVers.add(dvc.getDistributorVersion());
+        try {
+            return this.entityManager.get()
```
Comment (Ceiu) [inline src/main/java/org/candlepin/model/DistributorVersionCurator.java:50]: For future reference, you can call `getEntityManager` to skip the provider.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**